### DISCUSSION TITLE
Following multiple Reflections

### DISCUSSION THREAD

**Discussion by Tscho1995** [mitsuba-renderer/mitsuba3#760] [Q&A]
Hello all together, 

i have issues with understanding the BSDF's and multiple Reflections. I try to check if i get correct multiple reflections. For this purpose i implemented a little test environment.  

I create a really strong point light source and point it towards an object which is an ideal mirror(all red boxes in the image are mirrors) with  ggx distribution and alpha = 0.0, i see the first reflection, marked blue in the image. This is already suprising for me, since it shouldnt scatter. on the other mirror i see another reflection, marked purple. But this is the last reflection. so there are no more reflections.

My expaction was, that i dont see any reflections on the red parts in the image but on the yellow parts, since the red parts are ideal mirrors and the yellow part not.

I set the path depth to 8.

can someone explain me what i'm doing wrong? Thank you

Legend to image:
red boxes: ideal mirror, ggx distribution with alpha = 0.0
green box: ggx distribution with alpha = 0.05
blue: first reflection
purple: second reflection
yellow: where i expected to have the reflections

![reflexionen](https://github.com/mitsuba-renderer/mitsuba3/assets/72986938/14bcd041-a8e1-42d3-a645-407707c7c5de)

**Comment by njroussel** [comment]
Hi @Tscho1995 

Setting the roughness to 0 is not supported in Mitsuba3 IIRC, in fact, you should have gotten a warning for this ([source code](https://github.com/mitsuba-renderer/mitsuba3/blob/master/include/mitsuba/render/microfacet.h#L136-L139)). What you are most likely seeing is the small amount of clamped roughness.

For ideal mirrors you should use the [`conductor` plugin](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#smooth-conductor-conductor) with the `none` material as indicated in its documentation.

**Reply by njroussel** [reply]
P.S: I didn't mean to close the discussion... and apparently there is no option to "re-open" it :facepalm:

**Reply by Tscho1995** [reply]
Hello @njroussel,

thanks for your reply. I thought the `roughconductor `gets for really small alphas to `conductor `plugin. Sorry my mistake.

I did the rendering with `conductor` now. So red parts are conductor now, and indeed the reflexions on the red Blocks are gone, but still there is none in the yellow part. Yellow is `roughconductor ` with `alpha = 0.005`. 

Any suggestion what i'm doing wrong? Or how could i change my experiment?
![reflexionen_conductor](https://github.com/mitsuba-renderer/mitsuba3/assets/72986938/6e0300ac-bb26-4615-a3e2-5219e8d715c5)


Thank You

**Reply by njroussel** [reply]
I'm honestly a bit confused by your experiment/setup. From your images I can't really understand what the scene geometry looks like and where the emitters are.

So the yellow box is on the same plane (?) as the green box, correct? Maybe the emitters are just not visible for that angle? Keep in mind that if you're expecting to model a path as follows `eye -> rough surface -> perfect mirror -> emitter` this will never be possible with a point light as it's statistically impossible to connect the mirror to the emitter.

**Reply by Tscho1995** [reply]
Hello, i try to expain my set up. The Green Block is the whole block. And the yellow one is the side of it, see following image.
![grafik](https://github.com/mitsuba-renderer/mitsuba3/assets/72986938/978e5a09-ee2e-4792-ad23-1a9a7b50c7fe)

Now the next plot is what i imagine should happen. I'm know i won't see the yellow lines, but i expect to see one reflection on the green block which is marked, when i look from an angle on it.

![grafik](https://github.com/mitsuba-renderer/mitsuba3/assets/72986938/8a691979-f2e9-481c-bc0f-4cc463c289c4)

**Reply by njroussel** [reply]
Ok, I see. Thank you for clarifying.

What emitter plugin and integrator are you using? As I mentioned in my last sentence it is impossible to connect a perfect mirror (red box) to a point emitter if you start tracing rays from your camera. Instead, you should use an adjoint integrator like the [`ptracer`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#particle-tracer-ptracer) which starts tracing rays from the emitter and then tried to connect to the camera/sensor.

**Reply by Tscho1995** [reply]
I use a spot emitter and a path tracer with `depth=8`. Ok i understand with an perfect mirror that it is not possible. But which `alpha` value would suggest to achive my goal? 

Basically we have another industrial relevant use case, where we build a laserline as mitsuba emitter and we try to simulate real world problems with reflections on metal parts from laser line.

So we simulate the laser in Mitsuba and project it onto the part.  In this case we miss Reflections which appear in real world but not in the simulation. To fully understand the reflections i tried to do it with setup explained on top.

**Reply by njroussel** [reply]
Any reasonably small value should do. In fact, I'm a bit surprised that it wasn't visible in your original image. Maybe a higher `spp` value is also needed.
I would double-check your scene geometry, you should be able to analytically determine where the first bounce on the rough (green) surface should be. You can then simulate a single ray going through that path to confirm where it is ending up. 

Hope this helps.
